### PR TITLE
fix grappling bugs

### LINKED ISSUE #63: Grappling hook only fires to the right
# Description

We may want the grappling hook to fire to the left as well if the player faces the left

### PR DESCRIPTION
**Pull request by HoJunHao2000**
Resolves #63 

# Changes made
1. Players can now fire grappling hook to the right when facing right
2. For some reason circle physics objects are quite buggy for the hook so changed back to rectangle
3. Players no longer phase through obstacles when swinging. Will release when player collides with an obstacle.
4. Min length recalculated. Player will no longer swing (i.e. release) when hook collides with obstacle and there is insufficient space.
5. Players no longer swing in a large angle (i.e. swing 360 degrees when holding move input)
6. PositionComponent now has a isLeft variable

### CODE REVIEW

File: star-dash/star-dash/Constants/SpriteConstants.swift [hunk 1]
```diff
@@ -20,6 +20,7 @@ struct TextureSet {
 
 struct SpriteConstants {
     static let PlayerRedNose = "PlayerRedNose"
+    static let PlayerRedNoseLeft = "PlayerRedNose"
```
Comment (Junyi00) [inline star-dash/star-dash/Constants/SpriteConstants.swift:23]: What was this introduced for?

File: star-dash/star-dash/GameEngine/Systems/Complex/CollisionSystem.swift [hunk 2]
```diff
@@ -104,6 +104,12 @@ class CollisionSystem: System {
     }
 
     private func handlePlayerObstacleContactEvent(event: PlayerObstacleContactEvent) {
+        if let hookOwnerComponent = entityManager
+                                    .components(ofType: GrappleHookOwnerComponent.self)
+                                    .first(where: { $0.ownerPlayerId == event.playerId }) {
+            dispatcher?.add(event: ReleaseGrappleHookEvent(using: hookOwnerComponent.entityId))
```
Comment (Junyi00) [inline star-dash/star-dash/GameEngine/Systems/Complex/CollisionSystem.swift:110]: I'm quite conflicted about having any grapple-related logic here, and generally the purpose of collision system. It feels like `CollisionSystem` may fall into a "god" class problem where it's handling too many logic. For example , once we start having more features that rely on `PlayerObstacleContactEvent`, this single event listener will be doing way too much.

TLDR, I think this should be in `GrappleHookSystem` since its a grapple related event and we may want to start considering to phase out `CollisionSystem`

Wondering how @ruihan00 and @jasonqiu212 feels about this?

**Review by Junyi00 — COMMENTED** [review]
There's also an additional issue where players can walk in air after the rope releases (regardless of whether the hook was successful or not), but I believe it is an issue with the logic of our movement system rather than the grapple hook itself.

File: star-dash/star-dash/Constants/SpriteConstants.swift (hunk 1 above)
Comment (HoJunHao2000) [reply star-dash/star-dash/Constants/SpriteConstants.swift]: I iniitally wanted to the player to stay looking left after moving left, but didn't know how to and gave up.

Its irrelevant now. deleted.

File: star-dash/star-dash/GameEngine/Systems/Complex/CollisionSystem.swift (hunk 2 above)
Comment (HoJunHao2000) [reply star-dash/star-dash/GameEngine/Systems/Complex/CollisionSystem.swift]: Sound good to me. Move logic to GrappleHookSystem

**Review by Junyi00 — APPROVED** [review]
lgtm!
